libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/TemplatesRegisterExtensions.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveExpressions;
+using AdaptiveExpressions.Memory;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// Register lg templates into Expression.
+    /// Using prefix lg. to avoid the conflict with pre-built functions.
+    /// </summary>
+    public static class TemplatesRegisterExtensions
+    {
+        /// <summary>
+        /// Register Template functions.
+        /// </summary>
+        /// <param name="dialogContext">Dialog context.</param>
+        /// <param name="generator">Language generator.</param>
+        public static void RegisterTemplateFunctions(this DialogContext dialogContext, LanguageGenerator generator)
```

Comment:
>  public static void RegisterTemplateFunctions(this DialogContext dialogContext, LanguageGenerator generator) [](http://example.com/codeflow?start=6&length=109)

we should simply load all templates as functions into function tables but prefixed with file name (aka foo.lg.TEMPLATE)

Reply:
I will prefer to that would cause a "difference" in lg functions and other functions, which means the message to user could be still confusing, ie, they have to know, "OK, in sendActivity\prompts, i can call lg templates without prefix, in other actions, i have to do it with prefix".